src/main/groovy/com/twcable/gradle/sling/osgi/BundleAndServers.groovy

```diff
@@ -0,0 +1,553 @@
+     *
+     * @return the "aggregate" HTTP response: if all the calls are in the >= 200 and <400 range, or
+     * a 408 (timeout, server not running) the returns an empty HTTP_OK; otherwise returns the first error response
+     * it came across
+     */
+    @Nonnull
+    HttpResponse doAcrossServers(BundleServerAction serverAction) {
+        return doAcrossServers(serversConfiguration, slingBundleConfig, serverAction)
+    }
+
+    /**
+     * Runs the given server action across all the provided active servers
+     *
+     * @return the "aggregate" HTTP response: if all the calls are in the >= 200 and <400 range, or
+     * a 408 (timeout, server not running) the returns an empty HTTP_OK; otherwise returns the first error response
+     * it came across
+     */
+    @Nonnull
+    static HttpResponse doAcrossServers(SlingServersConfiguration servers,
+                                        SlingBundleConfiguration bundleConfiguration,
+                                        BundleServerAction serverAction) {
+        return doAcrossServers(servers, bundleConfiguration, SimpleSlingSupportFactory.INSTANCE, serverAction)
+    }
+
+    /**
+     * Runs the given server action across all the provided active servers
+     *
+     * @return the "aggregate" HTTP response: if all the calls are in the >= 200 and <400 range, or
+     * a 408 (timeout, server not running) the returns an empty HTTP_OK; otherwise returns the first error response
+     * it came across
+     */
+    @Nonnull
+    static HttpResponse doAcrossServers(SlingServersConfiguration servers,
+                                        SlingBundleConfiguration bundleConfiguration,
+                                        SlingSupportFactory slingSupportFactory,
+                                        BundleServerAction serverAction) {
+        def httpResponse = new HttpResponse(HTTP_OK, '')
+
+        servers.each { serverConfig ->
+            def slingSupport = slingSupportFactory.create(serverConfig)
+            def bundleServerConf = new BundleServerConfiguration(serverConfig)
+            def slingBundleSupport = new SlingBundleSupport(bundleConfiguration, bundleServerConf, slingSupport)
+            if (serverConfig.active) {
+                def resp = slingSupport.doHttp { SimpleHttpClient httpClient ->
+                    serverAction.run(slingBundleSupport)
+                }
+
+                // restrictive: if any call is bad, the result is bad
+                if (isBadResponse(resp.code)) {
+                    httpResponse = resp
+                }
+            }
+        }
+        return httpResponse
+    }
+
+
+    static boolean isBadResponse(int respCode) {
+        if (respCode >= HTTP_OK) {
+            if (respCode < HTTP_BAD_REQUEST) {
+                return false
+            }
+            return true
+        }
+        if (respCode == HTTP_CLIENT_TIMEOUT) return false
+        return true
```

Comment:
Look at the name of the function... ;-)

`HTTP_CLIENT_TIMEOUT` isn't considered "bad" in our use-case because it just means the server isn't there, not that something went wrong.